src/v2/segment.v2.ejs

```diff
@@ -0,0 +1,159 @@
+
+const SEGMENT_WRITE_KEY = "<%%= sd.SEGMENT_WRITE_KEY %>";
+const SEGMENT_MAPPING = 'C0001'; // Segment itself assigned OneTrust Strictly Necessary category.
+let SEGMENT_DESTINATIONS = [];
+
+fetchDestinations(SEGMENT_WRITE_KEY).then(
+  destinations => {
+    SEGMENT_DESTINATIONS = destinations;
+
+    waitForOneTrust();
+
+    if (oneTrustReady()) {
+      getConsentAndLoadSegment();
+    }
+  }
+)
+
+// OneTrust calls OptanonWrapper when there's a consent change.
+window.OptanonWrapper = getConsentAndLoadSegment;
+
+function waitForOneTrust() {
+  let attempts = 0;
+  const maxAttempts = 100;
+
+  // wait 1sec at most.
+  if (!oneTrustReady()) {
+    const interval = setInterval(() => {
+      if (oneTrustReady() || attempts > maxAttempts) clearInterval(interval);
+      attempts++;
+    }, 10);
+  }
+}
+
+function oneTrustReady() {
+    // OneTrust is ready if OnetrustActiveGroups contains at least Strictly Necessary group.
+    if ((typeof window.OnetrustActiveGroups === 'string') && (window.OnetrustActiveGroups.split(',').includes('C0001'))) {
+      return true;
+    }
+    return false;
+}
+
+async function fetchDestinations(writeKey) {
+  let destinations = await fetchDestinationForWriteKey(writeKey);
+
+  destinations = [
+    ...destinations
+      .reduce((map, item) => {
+        if (item.id === "Repeater") return map; // remove Segment Repeater destinations
+        map.has(item["id"]) || map.set(item["id"], item);
+        return map;
+      }, new Map()) // return object
+      .values()
+  ];
+
+  return destinations;
+}
+
+function getConsentAndLoadSegment() {
+  let oneTrustConsent = '';
+
+  // OneTrust stores consent in OnetrustActiveGroups.
+  oneTrustConsent = window.OnetrustActiveGroups.split(',');
+
+  const destinationPreferences = setSegmentDestinationPref(oneTrustConsent, SEGMENT_DESTINATIONS);
+
+  conditionallyLoadAnalytics({
+    writeKey: SEGMENT_WRITE_KEY,
+    destinations: SEGMENT_DESTINATIONS,
+    destinationPreferences
+  });
+}
+
+function setSegmentDestinationPref(oneTrustConsent, destinations) {
+  // maps Segment destination's category to OneTrust cookie category.
+  const segmentToOneTrust= {
+    'SMS & Push Notifications': 'C0001', // OneTrust Strictly Necessary
+    'Analytics': 'C0002', // OneTrust Performance
+    'Advertising': 'C0004' // OneTrust Targeting
+  }
+
+  const destinationPreferences = destinations
+    .map(function(dest) {
+      if ((dest.category in segmentToOneTrust) && oneTrustConsent.includes(segmentToOneTrust[dest.category])) {
+        return { [dest.id]: true };
+      } else {
+        // if we don't have a mapping for the Segment category, or we do but there's no OneTrust consent.
+        return { [dest.id]: false };
+      }
+    })
+    .reduce(
+      (acc, val) => {
+        return {
+          ...val,
+          ...acc
+        };
+      },
+      { "Segment.io": oneTrustConsent.some(d => d === SEGMENT_MAPPING) }
+    );
+
+  return destinationPreferences;
```

Comment:
Example of its content:
<img width="458" alt="Screen Shot 2021-11-03 at 4 54 57 PM" src="https://user-images.githubusercontent.com/63004951/140980251-b1442f20-045e-4dfc-9bef-9b8b1138fbc4.png">